src/cli/install.ps1

```diff
@@ -1,35 +1,95 @@
 #!/usr/bin/env pwsh
 param(
   # TODO: change this to 'latest' when Bun for Windows is stable.
-  [string]$Version = "canary"
+  [String]$Version = "canary",
+  # Forces installing the baseline build regardless of what CPU you are actually using.
+  [Switch]$ForceBaseline = $false,
+  # Skips adding the bun.exe directory to the user's %PATH%
+  [Switch]$NoPathUpdate = $false,
+  # Skips adding the bun to the list of installed programs
+  [Switch]$NoRegisterInstallation = $false
 );
 
+# filter out 32 bit + ARM
+if ($env:PROCESSOR_ARCHITECTURE -ne "AMD64") {
+  Write-Output "Install Failed:"
+  Write-Output "Bun for Windows is only available for x86 64-bit Windows.`n"
+  exit 1
+}
+
+# This corresponds to .win10_rs5 in build.zig
+$MinBuild = 17763;
+$MinBuildName = "Windows 10 1809"
+$WinVer = [System.Environment]::OSVersion.Version
+if ($WinVer.Major -lt 10 -or ($WinVer.Major -eq 10 -and $WinVer.Build -lt $MinBuild)) {
+  Write-Warning "Bun requires at $($MinBuildName) or newer.`n`nThe install will still continue but it may not work.`n"
```

Comment:
The `$($MinBuildName)` seems a little overkill, I think it should be just `$MinBuildName` or instead place the `Windows 10 1809` text there (as its the only usage).